buildSrc/src/main/java/quilt/internal/tasks/decompile/DecompileTask.java

```diff
@@ -23,89 +24,69 @@
-    private final RegularFileProperty output;
-    private final Property<FileCollection> libraries;
+public abstract class DecompileTask extends DefaultMappingsTask {
+    @Input
+    public abstract Property<Decompilers> getDecompiler();
+
+    @Optional
+    @Input
+    public abstract MapProperty<String, Object> getDecompilerOptions();
+
+    @Optional
+    @Input
+    public abstract Property<FileCollection> getLibraries();
+
+    @InputFiles
+    public abstract RegularFileProperty getInput();
+
+    @OutputDirectory
+    public abstract RegularFileProperty getOutput();
+
+    // TODO see if all three of these can be optional inputs
     private ClassJavadocProvider classJavadocProvider;
     private FieldJavadocProvider fieldJavadocProvider;
     private MethodJavadocProvider methodJavadocProvider;
 
     public DecompileTask() {
         super(Constants.Groups.DECOMPILE_GROUP);
-        input = getProject().getObjects().fileProperty();
-        output = getProject().getObjects().fileProperty();
-        decompiler = getProject().getObjects().property(Decompilers.class);
-        libraries = getProject().getObjects().property(FileCollection.class);
     }
 
     @TaskAction
     public void decompile() {
-        Map<String, Object> options = decompilerOptions == null ? new HashMap<>() : decompilerOptions;
-        Collection<File> libraries = this.libraries.getOrNull() == null ? Collections.emptyList() : this.libraries.get().getFiles();
+        final Map<String, Object> options = this.getDecompilerOptions().getOrElse(new HashMap<>());
 
-        AbstractDecompiler decompiler = getAbstractDecompiler();
+        final Collection<File> libraries = this.getLibraries().<Collection<File>>map(FileCollection::getFiles)
+            .getOrElse(Collections.emptyList());
 
-        if (classJavadocProvider != null) {
-            decompiler.withClassJavadocProvider(classJavadocProvider);
+        final AbstractDecompiler decompiler = this.getAbstractDecompiler();
+
+        if (this.classJavadocProvider != null) {
+            decompiler.withClassJavadocProvider(this.classJavadocProvider);
         }
-        if (fieldJavadocProvider != null) {
-            decompiler.withFieldJavadocProvider(fieldJavadocProvider);
+        if (this.fieldJavadocProvider != null) {
+            decompiler.withFieldJavadocProvider(this.fieldJavadocProvider);
         }
-        if (methodJavadocProvider != null) {
-            decompiler.withMethodJavadocProvider(methodJavadocProvider);
+        if (this.methodJavadocProvider != null) {
+            decompiler.withMethodJavadocProvider(this.methodJavadocProvider);
         }
 
-        decompiler.decompile(getInput().getAsFile().get(), getOutput().getAsFile().get(), options, libraries);
+        decompiler.decompile(this.getInput().getAsFile().get(), this.getOutput().getAsFile().get(), options, libraries);
     }
 
     @Internal
     public AbstractDecompiler getAbstractDecompiler() {
```

Comment:
Similar to the EnigmaProfile stuff above -- this is not conf cache compatible, and looks like a perfect place to use a build service.